### ISSUE TITLE
If a Default RelayState is declared a passed RelayState is ignored on an unsolicited login. 

### ISSUE THREAD

**Issue by jeremyfrench** [simplesamlphp/simplesamlphp#1296]
**Describe the bug**
I have recently done a big upgrade of SimpleSamlPHP from 1.11 to 1.18.4 generally things work as expected, but we experienced an unexpected change in behaviour with unsolicited logins and redirects. 

**To Reproduce**
1. In your auth source have a line for a default relay state `'RelayState' => '/saml_default',`
2. Do an unsolicited login from your IDP with a different relay state.

An example of the code in authsources.php is 

```
  $config[test1] = [
    'saml:SP',
    'entityID' => 'test1',
    'idp' => 'testidp'.
    'RelayState' => '/saml_login',
  ];
```

**Expected behavior**
I'd expect after login to be sent to the page that was sent in the request from the IDP if one is present and if it is not to be sent to the default, but the user is always sent to the default relay state.

As a workaround, I have removed the default and the RelayState then works as expected, this is fine for me as currently, it's fine to have the root page as the default. 

**Additional context**
I found https://github.com/simplesamlphp/simplesamlphp/issues/99 which seems to be related but this does not seem to be quite the same issue.

It could be that the RelayState in here should always be used and that 1.11 was behaving incorrectly but I could find scant documentation on this config option. I have looked at the changelog and can't see anything that looks obviously related to indicate a change in behaviour.

If this is actually a bug, I'm happy to look at writing a fix, but I'm not 100% sure if the old behaviour is expected or not.

**Comment by jaimeperez** [comment]
Hi @jeremyfrench!

Thanks for reporting this! It looks like a bug to me. In fact, the standard mandates that the RelayState should be returned by an IdP verbatim, so we cannot return something else if we were given one. A fix would be very much welcome indeed!

**Comment by areis422** [comment]
Bump
Possibly related to #1473

**Comment by tvdijen** [comment]
Fixed in 96375af5606ad0cf035ccdae069bb4992b017ddc
